modules/browsemap/src/main/java/com/sygic/maps/module/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -131,59 +132,99 @@ class BrowseMapFragmentViewModel internal constructor(
             return
         }
 
+        // Currently, we take care only of the first ViewObject
         var firstViewObject = viewObjects.first()
+        val dataLoadAllowed = onMapClickListener?.onMapClick(firstViewObject) ?: true
+
+        // First, check if the PoiDetailsView is visible
         poiDetailsView?.let {
+
+            // Always remove the previous one
             mapDataModel.removeMapObject(it)
             poiDetailsView = null
-            if (firstViewObject !is MapMarker) {
-                return
+
+            // Check if we should ask or use our own logic
+            if (onMapClickListener != null) {
+                // This is useful for switching between the same types without hiding the previous one
+                if (!dataLoadAllowed) {
+                    return
+                }
+            } else {
+                // Currently, we internally support only MapMarker's
+                if (firstViewObject !is MapMarker) {
+                    return
+                }
             }
         }
 
         when (mapSelectionMode) {
+
             MapSelectionMode.NONE -> {
                 logWarning("NONE")
             }
+
             MapSelectionMode.MARKERS_ONLY -> {
+                // Markers only mode check
                 if (firstViewObject !is MapMarker) {
                     return
                 }
 
+                // Continue only if the ViewObject data load is allowed
+                if (!dataLoadAllowed) {
+                    return
+                }
+
                 getPoiDataAndNotifyObservers(firstViewObject)
             }
+
             MapSelectionMode.FULL -> {
-                if (firstViewObject !is MapMarker && onMapClickListener == null) {
-                    mapDataModel.addOnClickMapMarker(when (firstViewObject) {
-                        is ProxyPoi -> MapMarker.from(firstViewObject).build()
-                        else -> MapMarker.at(firstViewObject.position).build()
-                    }.also { firstViewObject = it })
+                // Add the OnClickMapMarker only if the click is not at MapMarker
+                if (firstViewObject !is MapMarker) {
+                    getOnClickMapMarker(firstViewObject)?.let { clickMapMarker ->
+                        mapDataModel.addOnClickMapMarker(
+                            when (firstViewObject) {
+                                is ProxyPoi -> MapMarker.at(clickMapMarker.position).withPayload(firstViewObject).build()
```

Comment:
ked je to proxyPoi, tak odignorujeme marker od usera?

Reply:
asi ano, ještě se na to podívám. toto je ten "hack" kvůli měření toho odsazení pro DetailsView :(

Reply:
tak opraveno ;)